packages/common/src/audius-query/types.ts

```diff
@@ -56,6 +56,13 @@ export type Api<EndpointDefinitions extends DefaultEndpointDefinitions> = {
   fetch: {
     [Property in keyof EndpointDefinitions]: EndpointDefinitions[Property]['fetch']
   }
+  fetchSaga: {
+    [Property in keyof EndpointDefinitions]: (
+      fetchArgs: any,
```

Comment:
it would be cool to get the type at least for the args working, and ideally the return type

these are tough though

Reply:
+1

Reply:
I spent some time trying to get the args working but kept running into the `could be instantiated with an arbitrary type` issue. I had the same problem trying to type the args for `fetchBatch`, I think we may be running into the boundaries of what typescript can handle tbh